extruct/xmldom.py

```diff
@@ -47,7 +41,7 @@ def lxmlDomNodeType(node):
     if isinstance(node, ElementBase):
         return Node.ELEMENT_NODE
 
-    elif isinstance(node, (_ElementStringResult, _ElementUnicodeResult)):
+    elif isinstance(node, _ElementUnicodeResult):
```

Comment:
This would require a baseline dependency of `lxml>=5.1.1`, but perhaps we could use `node.is_text` and `node.is_attribute` instead of the conditions within this code block?

Reply:
I added a dummy class on import error since it doesn't exist in newer versions anyway, which should allow for full backwards compatibility